grin/src/AbstractInterpretation/CreatedBy/CodeGenBase.hs

```diff
@@ -126,8 +126,8 @@ codeGenBlock_ = fmap snd . codeGenBlock
 codeGenSimpleType :: SimpleType -> CG IR.Reg
 codeGenSimpleType = \case
   T_Unit                -> newRegWithSimpleType (-1)
-  T_Int64               -> newRegWithSimpleType (-2)
-  T_Word64              -> newRegWithSimpleType (-3)
+--  T_Int width           -> newRegWithSimpleType (-2)
```

Comment:
@csabahruska do you have an existing plan how to implement this in these abstract interpretations?

Reply:
I was wondering a bit while I was writing this why a mapping from types to registers wasn't handled by `Data.Map`, maybe that's a solution, to give unique types the next available register?